### PR TITLE
show Journal Size in In-Place Migration steps

### PR DESCRIPTION
**Pull request by gally47**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/datanode/hooks/useJournalDowntimeSize.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  refetch: () => void,
+  isInitialLoading: boolean,
+  error: any,
+} => {
+  const { data, refetch, isInitialLoading, error } = useQuery(
+    ['JournalDowntimeSize'],
+    fetchJournalDowntimeSize,
+    {
+      onError: (errorThrown) => {
+        UserNotification.error(`Loading Data Node failed with status: ${errorThrown}`,
```
Comment (ousmaneo) [inline graylog2-web-interface/src/components/datanode/hooks/useJournalDowntimeSize.ts:36]:
```suggestion
        UserNotification.error(`Loading Data Node migration journal estimate failed with status: ${errorThrown}`,
```

File: graylog2-web-interface/src/components/datanode/hooks/useJournalDowntimeSize.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          'Could not load Data Node');
```
Comment (ousmaneo) [inline graylog2-web-interface/src/components/datanode/hooks/useJournalDowntimeSize.ts:37]:
```suggestion
          'Could not load Data Node journal size estimate.');
```

File: graylog2-web-interface/src/components/datanode/migrations/in-place/JournalDowntimeWarning.tsx [hunk 3]
```diff
@@ -21,21 +21,28 @@ import { Alert } from 'components/bootstrap';
...
-);
+const JournalDowntimeWarning = ({ currentStep, onTriggerStep }: MigrationStepComponentProps) => {
+  const { data } = useJournalDowntimeSize();
+
+  return (
+    <>
+      <h3>Journal downtime size warning</h3>
+      <p>Please note that during migration data processing  will stop on your Graylog node, this will result in the journal growing in size.</p>
+      <p>Therefore increase your journal volume size.</p>
+      <p>Size: <b>{data} KB/min</b></p>
```
Comment (ousmaneo) [inline graylog2-web-interface/src/components/datanode/migrations/in-place/JournalDowntimeWarning.tsx:39]:
```suggestion
      <p>Your current journal throughput is: <b>{data} KB/min</b></p>
```

File: graylog2-web-interface/src/components/datanode/migrations/in-place/JournalDowntimeWarning.tsx [hunk 4]
```diff
@@ -21,21 +21,28 @@ import { Alert } from 'components/bootstrap';
...
-    <p>Please note that during migration data processing  will stop on your Graylog node, this will result in the journal growing in size.</p>
-    <p>Therefore increase your journal volume size.</p>
-    <DownsizeWarning bsStyle="danger">
-      <p>Please make sure your journal volume size is sufficient before proceeding.</p>
-    </DownsizeWarning>
-    <MigrationStepTriggerButtonToolbar nextSteps={currentStep.next_steps} onTriggerStep={onTriggerStep} />
-  </>
-);
+const JournalDowntimeWarning = ({ currentStep, onTriggerStep }: MigrationStepComponentProps) => {
+  const { data } = useJournalDowntimeSize();
```
Comment (ousmaneo) [inline graylog2-web-interface/src/components/datanode/migrations/in-place/JournalDowntimeWarning.tsx:32]: We should also handle displaying a message when there is an error when getting the throughput estimation.

**Review by ousmaneo — CHANGES_REQUESTED** [review]
Looks good, I added some suggestion that we should take care of.

**Review by ousmaneo — APPROVED** [review]
LGTM!

**Review by moesterheld — APPROVED** [review]
lgtm

if we get feedback from testers about how long they took on average for the restart, we could add a estimate for the journal size that multiplies this values by the needed number of minutes (plus buffer)

**Pull request merged**